frontend/src/utils/generate-warnings.ts

```diff
@@ -184,6 +184,9 @@ function produceUnsatifiedRequirement(
   taken: Map<string, HashableCourse>,
   coursesUsed: Set<string>
 ): IRequirementGroupWarning | undefined {
+  if (!requirementGroup) {
+    return undefined;
+  }
```

Comment:
Why did you make this change? I just deployed after removing it, and it worked fine.

Reply:
You're right! I think I came upon an error before where it was undefined, but I think I was doing something else wrong.